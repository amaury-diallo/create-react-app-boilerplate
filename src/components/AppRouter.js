import React from 'react'

import { Switch } from 'react-router'
import { Router, Route, Redirect } from 'react-router-dom'
import history from '../config/initializers/history'

import { Me, About } from '../containers'

const AppRouter = () =>
  <Router history={history}>
    <Switch>
      <Route path="/me" component={Me} />
      <Route path="/about" component={About} />
      <Redirect path="*" to="/me" />
    </Switch>
  </Router>

export default AppRouter
