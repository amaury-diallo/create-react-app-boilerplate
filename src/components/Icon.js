import React from 'react'
import PropTypes from 'prop-types'

Icon.propTypes = {
  name: PropTypes.string.isRequired
}

function Icon(props) {
  const className = `fa fa-${props.name}`

  return (
    <i className={className}></i>
  )
}

export default Icon
