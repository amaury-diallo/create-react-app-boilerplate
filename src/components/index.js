import AppRouter from './AppRouter'
import Icon from './Icon'

export {
  AppRouter,
  Icon
}
