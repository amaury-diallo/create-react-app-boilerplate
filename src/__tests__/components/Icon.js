import { Icon } from '../../components'

describe('<Icon />', () => {
  const props = {
    name: 'plus'
  }

  let wrapper
  beforeEach(() => {
    wrapper = shallow(<Icon {...props} />)
  })

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })
  it('should have the right class name', () => {
    expect(wrapper.props().className).toEqual('fa fa-plus')
  })
})