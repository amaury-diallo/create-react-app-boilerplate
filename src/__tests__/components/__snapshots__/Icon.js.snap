// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Icon
    name="plus"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "fa fa-plus",
    },
    "ref": null,
    "rendered": null,
    "type": "i",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "fa fa-plus",
      },
      "ref": null,
      "rendered": null,
      "type": "i",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
